src/Serilog.Sinks.AzureDataExplorer.Tests/AzureDataExplorerSinkOptionsExtensionsTests.cs

```diff
@@ -0,0 +1,137 @@
+using System.Security.Cryptography.X509Certificates;
+using System.Security.Cryptography;
+using Serilog.Sinks.AzureDataExplorer.Extensions;
+
+namespace Serilog.Sinks.AzureDataExplorer.Tests;
+
+public class AzureDataExplorerSinkOptionsExtensionsTests
+{
+    [Fact]
+    public void GetKustoConnectionStringBuilder_AadApplicationKey_ReturnsExpectedConnectionString()
+    {
+        var options = new AzureDataExplorerSinkOptions
+        {
+            IngestionEndpointUri = "https://ingestion-endpoint-uri",
+            DatabaseName = "database-name",
+            TableName = "table-name"
+        }.WithAadApplicationKey("applicationClientId", "applicationClientId", "authority");
+
+        var kcsb = options.GetKustoConnectionStringBuilder();
+        var connectionString = kcsb.ToString();
+        Assert.NotNull(kcsb);
+        Assert.NotEmpty(connectionString);
+    }
+
+    [Fact]
+    public void GetKustoConnectionStringBuilder_AadApplicationCertificate_ReturnsExpectedConnectionString()
```

Comment:
You can collapse the tests below with [Theory] / Inline data if you want , less code more easier to read